src/masternodes/mn_checks.cpp

```diff
@@ -1029,6 +1079,406 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
+
+        if (order.orderType == 0) {
+            return Res::Err("invalid order type!");
+        }
+        if (order.orderType == CICXOrder::TYPE_INTERNAL)
+        {
+            auto tokenFrom = mnview.GetToken(order.idToken);
+            if (!tokenFrom) {
+                return Res::Err("%s: %s", __func__, "tokenFrom (" +  tokenFrom->CreateSymbolKey(order.idToken) + ") does not exist!");
+            }
+
+            if (order.ownerAddress.empty())
+                return Res::Err("ownerAddress must be not-null!");
+        }
+        else
+        {
+            auto tokenTo = mnview.GetToken(order.idToken);
+            if (!tokenTo) {
+                            return Res::Err("%s: %s", __func__, "tokenTo id (" +  tokenTo->CreateSymbolKey(order.idToken) + ") does not exist!");
+            }
+        }
+
+        if (order.amountFrom == 0) {
+            return Res::Err("order amountFrom must be greater than 0!");
+        }
+        if (order.orderPrice == 0) {
+            return Res::Err("order price must be greater than 0!");
+        }
+        if (order.amountToFill != order.amountFrom*COIN/order.orderPrice) {
+            return Res::Err("order amountToFill does not equal to amountFrom * orderPrice!");
+        }
+        if (order.expiry == 0) {
+            return Res::Err("order expiry must be greater than 0!");
+        }
+
+        // subtract the balance from tokenFrom to dedicate them for the order
+        if (order.orderType == CICXOrder::TYPE_INTERNAL)
+        {
+            CTokenAmount amount={order.idToken,order.amountFrom};
+            auto res = mnview.SubBalance(order.ownerAddress,amount);
+            if (!res.ok) {
+                return Res::Err("%s: %s", __func__, res.msg);
+            }
+        }
+
+        order.creationTx = tx.GetHash();
+        order.creationHeight = height;
+        order.expireHeight = order.creationHeight + order.expiry;
+
+        return mnview.ICXCreateOrder(order);
+    }
+
+    Res operator()(const CICXMakeOfferMessage& obj) const {
+        auto res = CheckICXTx();
+        if (!res) {
+            return res;
+        }
+
+        CICXMakeOfferImplemetation makeoffer;
+        static_cast<CICXMakeOffer&>(makeoffer) = obj;
+
+        auto order=mnview.GetICXOrderByCreationTx(makeoffer.orderTx);
+        if (!order) {
+            return Res::Err("%s: %s", __func__, "order with creation tx " + makeoffer.orderTx.GetHex() + " does not exists!");
+        }
+
+        if (order->orderType == CICXOrder::TYPE_INTERNAL)
+        {
+            if (makeoffer.receiveDestination.empty())
+                return Res::Err("receiveAddress must not be null!");
+        }
+        else
+        {
+            if (!CPubKey(makeoffer.receiveDestination).IsFullyValid())
+                return Res::Err("%s: %s", __func__, "Invalid receivePubKey, (" + HexStr(makeoffer.receiveDestination) + ") receivePubkey is not a valid pubkey!");
```

Comment:
Remove all reference to `__func__`. It's not needed anymore, just return a string it will be constructed for concrete function.